sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2244 @@
+      std::vector<uint8_t> Body; // Initial data
+      std::string ClientRequestId = std::string(); // Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the analytics logs when storage analytics logging is enabled.
+      std::string ApiVersionParameter = k_DefaultServiceApiVersion; // Specifies the version of the operation to use for this request.
+    };
+
+    static Azure::Core::Http::Request PathAppendDataCreateRequest(std::string url, const PathAppendDataOptions& pathAppendDataOptions)
+    {
+      Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Patch, std::move(url), pathAppendDataOptions.Body);
+      request.AddQueryParameter(k_QueryAction, "append");
+      // TODO: This should be changed to null check when Optional<T> is ready.
+      if (!std::to_string(pathAppendDataOptions.Position).empty())
+      {
+        request.AddQueryParameter(k_QueryPosition, std::to_string(pathAppendDataOptions.Position));
+      }
+      // TODO: This should be changed to null check when Optional<T> is ready.
+      if (!std::to_string(pathAppendDataOptions.Timeout).empty())
+      {
+        request.AddQueryParameter(k_QueryTimeout, std::to_string(pathAppendDataOptions.Timeout));
+      }
+      // TODO: This should be changed to null check when Optional<T> is ready.
+      if (!std::to_string(pathAppendDataOptions.ContentLength).empty())
+      {
+        request.AddHeader(k_HeaderContentLength, std::to_string(pathAppendDataOptions.ContentLength));
+      }
+      if (!pathAppendDataOptions.TransactionalContentMD5.empty())
+      {
+        request.AddHeader(k_HeaderTransactionalContentMD5, pathAppendDataOptions.TransactionalContentMD5);
+      }
+      if (!pathAppendDataOptions.LeaseIdOptional.empty())
+      {
+        request.AddHeader(k_HeaderLeaseIdOptional, pathAppendDataOptions.LeaseIdOptional);
+      }
+      if (!pathAppendDataOptions.ClientRequestId.empty())
+      {
+        request.AddHeader(k_HeaderClientRequestId, pathAppendDataOptions.ClientRequestId);
+      }
+      request.AddHeader(k_HeaderApiVersionParameter, pathAppendDataOptions.ApiVersionParameter);
+      return request;
+    }
+
+    struct PathAppendDataResponse
+    {
+      std::string Date;
+      std::string RequestId;
+      std::string ClientRequestId;
+      std::string Version;
+    };
+
+    static PathAppendDataResponse PathAppendDataParseResponse(/*const*/ std::unique_ptr<Azure::Core::Http::Response>& responsePtr)
+    {
+      /* const */ auto& response = *responsePtr;
+      if (response.GetStatusCode() == Azure::Core::Http::HttpStatusCode::Accepted)
+      {
+        // Append data to file control response.
+        PathAppendDataResponse result;
+        result.Date = response.GetHeaders().at(k_HeaderDate);
+        result.RequestId = response.GetHeaders().at(k_HeaderXMsRequestId);
+        result.ClientRequestId = response.GetHeaders().at(k_HeaderXMsClientRequestId);
```

Comment:
client request id is optional, it may not appear in response headers.

Reply:
Made all response header check existence before getting the value.

Reply:
not ideal but fine.